src/components/SwapWidget/subcomponents/InfoSection/InfoSection.tsx

```diff
@@ -116,6 +122,104 @@ const InfoSection: FC<InfoSectionProps> = ({
     );
   }
 
+  // if custom server is selected, display custom server message
+  if (hasSelectedCustomServer && !isPairUnavailable && !bestTradeOption) {
```

Comment:
There are way too many changes for one simple message. Also I see you added nested if's, don't do that. One if for every state.

Reply:
I removed the nested if. Now I see that it wasn't necessary. There aren't many actual changes to the code, but it appears that way because I moved everything into the nested 'if' statement (which has since been removed)